pyopencl/array.py

```diff
@@ -2579,6 +2579,38 @@ def hstack(arrays, queue=None):
 
     return result
 
+
+def stack(arrays, axis=0, queue=None):
+    if not arrays:
+        raise ValueError("need at least one array to stack")
+
+    if axis != 0:
+        # pyopencl.Array.__setitem__ does not support non-contiguous assignments
+        raise NotImplementedError("axis!=0 not implemented")
```

Comment:
This is only true for C-layout arrays. (`ary.flags.c_contiguous`) So IMO this needs to be checked.

Reply:
Yep, wasn't correct for all array layout orderings. Fixed in ad6b6957a0fef0e65b55660311d5830a2c72e8b5.